### PR TITLE
help: display with the organization's code

### LINKED ISSUE #709: Help contents should not redirect to SONAR global for dedicated repositories [3]
In dedicated repositories, the help link switches to a SONAR-like interface, preventing to come back to the dedicated interface.

### LINKED ISSUE #712: Bugs in the help editor: impossible to upload a file, impossible to preview a page in the editor [0]

### PR DESCRIPTION
**Pull request by Garfield-fr**
The help is displayed with the organization's code
and is no longer directed to the global help.

* Closes #709.
* Closes #712.

Co-Authored-by: Bertrand Zuchuat <bertrand.zuchuat@rero.ch>

## How to test?

- Url for organization help: https://sonar/{view}/help

## Code review check list

- [ ] Commit message template compliance.
- [ ] Commit message without typos.
- [ ] File names.
- [ ] Functions names.
- [ ] Functions docstrings.
- [ ] Unnecessary commited files?
- [ ] Extracted translations?

### CODE REVIEW

File: sonar/dedicated/__init__.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
+# -*- coding: utf-8 -*-
+#
+# Swiss Open Access Repository
+# Copyright (C) 2021 RERO
```
Comment (jma) [inline sonar/dedicated/__init__.py:4]: 2022?

File: sonar/help/templates/help/page_wiki_base.html [hunk 2]
```diff
@@ -0,0 +1,36 @@
+{# -*- coding: utf-8 -*-
```
Comment (jma) [inline sonar/help/templates/help/page_wiki_base.html:1]: Why not using the existing template `sonar/page_wiki.html`?

File: sonar/help/templates/help/page_wiki_base.html [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+{{ super() }}
+<link href="{{ url_for('wiki.static', filename='css/wiki.css') }}" rel="stylesheet" />
+{%- endblock %}
+
+{%- block body %}
+
+{% block navigation %}
+{%- include "help/page_wiki_search_navigation.html" %}
+{%- endblock navigation %}
+
```
Comment (jma) [inline sonar/help/templates/help/page_wiki_base.html:34]: Javascript block seems missing.

File: sonar/help/templates/help/page_wiki_org.html [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+                {% endfor %}
+            </ul>
+            {% endif %}
+            <div class="text-muted small">
+                <span>{{ _('Created:')}} {{ page.creation_datetime | date_format }} - {{ _('Last updated:')}} {{ page.modification_datetime | date_format }}</span>
+            </div>
+            </footer>
+        </header>
+        <article>
+          {{ page.body.replace('](/', '](/{view}/'.format(view=view))|markdown_filter|safe }}
```
Comment (jma) [inline sonar/help/templates/help/page_wiki_org.html:42]: Is it really safe? Maybe we can replace only the `/help` links?

File: sonar/help/templates/help/page_wiki_search.html [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+                </div>
+        </header>
+        <article>
+            <ul class="list-group list-group-flush">
+                {%- for result in results -%}
+                <li class="list-group-item">
+                    <article class="d-lg-flex justify-content-lg-between">
+                        <article class="mb-2">
+                            <h5 class="m-0">
+                                <a class="mr-2" href="{{ url_for('help.page', view=view, url=result.url) }}">
```
Comment (jma) [inline sonar/help/templates/help/page_wiki_search.html:39]: I imagine this is the only modification done form flask-wiki. We can perhaps create a link factory for this?

File: sonar/help/views.py [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+
+
+@blueprint.route('/<org_code:view>/help/', methods=['GET'])
+def index(view):
+    """Help index redirect to home."""
+    return redirect(url_for(
+        'help.page', view=view, url=current_app.config.get('WIKI_HOME')))
+
+
+@blueprint.route('/<org_code:view>/help/<path:url>/', methods=['GET'])
```
Comment (jma) [inline sonar/help/views.py:37]: Please remove the trailing '/'.

**Review by jma — CHANGES_REQUESTED** [review]

**Comment by rerowep** [comment]
We are in 2022 new, do we have to change the copyright date in the changed file?

**Review by jma — APPROVED** [review]

**Comment by pronguen** [comment]
It works if we have the link.
But there is a default link "Aide et documentation" in sonardev, which always redirects on the global help (see screenshot). Is it possible to have the default link on the help with organisation view?
image

**Review by mmo — APPROVED** [review]

**Pull request merged**